src/CoreWCF.NetTcp/src/CoreWCF/Channels/Framing/ServerFramingDuplexSessionMiddleware.cs

```diff
@@ -86,15 +86,23 @@ public async Task OnConnectedAsync(FramingConnection connection)
                                 break;
 
                             case ServerSessionDecoder.State.Start:
-                                SetupSecurityIfNecessary(connection);
+                                try
+                                {
+                                    SetupSecurityIfNecessary(connection);
 
-                                // we've finished the preamble. Ack and continue to the next middleware.
-                                await connection.Output.WriteAsync(ServerSessionEncoder.AckResponseBytes);
-                                await connection.Output.FlushAsync();
-                                connection.Input.AdvanceTo(buffer.Start);
-                                success = true;
-                                await _next(connection);
-                                return;
+                                    // we've finished the preamble. Ack and continue to the next middleware.
+                                    await connection.Output.WriteAsync(ServerSessionEncoder.AckResponseBytes);
+                                    await connection.Output.FlushAsync();
+                                    connection.Input.AdvanceTo(buffer.Start);
+                                    success = true;
+                                    await _next(connection);
+                                    return;
+                                }
+                                catch(Exception ex)
+                                {
+                                    success = false;
```

Comment:
>success [](http://example.com/codeflow?start=36&length=7)

Nice find, but the try/catch/rethrow isn't necessary. Just move the `success = true;` until after the `await _next(connection);` The finally won't run until after the `_next` completes so moving setting success to true until after means any exceptions thrown will result in it being false in the finally block.  
Also FYI, rethrowing an exception like that results in the call stack in the exception being replaced with the current call stack. You can use `throw;` without specifying the exception and it will rethrow the current exception without modifying the throwing call stack. That then means you don't need to declare the ex parameter on the catch and can just do `catch(Exception)`. But as I said, try/catch not needed in this case.